code/frameworks/vue3-vite/src/plugins/vue-component-meta.ts

```diff
@@ -133,6 +134,19 @@ async function createVueComponentMetaChecker(tsconfigPath = 'tsconfig.json') {
   const checkerOptions: MetaCheckerOptions = {
     forceUseTs: true,
     noDeclarations: true,
+    schema: {
+      ignore: [
+        (name, type, typeChecker) => {
+          // Omit all schemas whose declaration paths are in node_modules and are objects.
+          const isInNodeModules = type.getSymbol()?.getDeclarations()?.some((declaration) => {
+            return declaration.getSourceFile().fileName.includes('node_modules')
```

Comment:
I am not sure if we can/should do it like that because that would mean the types are ignored for every node module.

So if you e.g. have a shared library where you import types from, they all will be ignored which would not be ideal and could potential break docs of existing projects that uses this.

Should we maybe only check for types of the TypeScript lib itself or let the user configure this via a framework option so its totally up to them which types they want to have excluded if they run into a performance issue?

Reply:
Even then there might be edge cases like: `Pick<HTMLElement, "propA" | "propB">` which does not cause memory issues but would be ignored this way

Reply:
Thank you for your thorough review and the points you've raised. I appreciate the opportunity to clarify and discuss these concerns. Let me address each point:

1. **Ignoring types for all node modules:**
   I understand your concern about potentially ignoring types from shared libraries or other important modules. However, the current implementation is specifically designed to address performance issues while maintaining functionality. As mentioned in the PR description, for Storybook's specific use case, object type schemas don't significantly contribute to generating better argsTypes.

2. **Storybook's handling of Object type schemas:**
   From my understanding of the Storybook documentation (https://storybook.js.org/docs/essentials/controls#annotation), there isn't a specialized control element for Object type schemas beyond a JSON-based editor. This editor doesn't seem to benefit from having detailed object prop schemas. If my interpretation of the documentation is incorrect, please let me know.

3. **Edge case: `Pick<HTMLElement, "propA" | "propB">`:**
   In this scenario, the resulting type is still an Object type. Given our current approach, ignoring the schema information for this case should not pose issues, as it aligns with our handling of other Object types.

4. **Handling of external type references:**
   For cases like `HTMLElement['id']`, our implementation would still provide the corresponding string schema information. This approach ensures that we're not losing critical type information for primitive types or those referenced from external sources.

5. **Impact on projects using external type references:**
   Based on the points above, I believe the current implementation shouldn't cause problems for projects that use external type references. We're primarily optimizing for Object types while preserving other critical type information.

That being said, I'm open to discussing alternative approaches:

1. We could consider limiting this optimization to TypeScript lib types only, as you suggested.
2. Alternatively, we could implement this as a configurable framework option, allowing users to specify which types they want to exclude if they encounter performance issues.

I'm happy to explore these options or any other suggestions you might have to find the best balance between performance optimization and maintaining robust type support. Your insights on this matter are valuable, and I'm eager to refine this approach to best serve the Storybook community.

Reply:
@et84121 I investigated the issue a little, it seems like `vue-component-meta` is not the issue here since it extracts the meta successfully. But in some place we process the data further within Storybook which seems to maybe have some recursive logic which causes the out of memory error.

You can verify this by changing line 88-120 in the `vue-component-meta.ts` file like this:

```ts
let s = new MagicString(src);
// ...
s = new MagicString(src);
return {
  code: s.toString(),
  map: s.generateMap({ hires: true, source: id }),
};
```

This will essentially clear any docgen information from further Storybook processing. The key point here is that the code runs successfully until this point so the `vue-component-meta` extraction works totally file.

Ideally we want to check which Storybook code causes the issue and try to fix it there instead of limiting the capabilities of `vue-component-meta` which could cause unintended behavior in projects :)

What do you think?

Reply:
It seems like there might be a circular reference within the args of a specific story objects. I recall encountering a similar issue before; it could be an edge case that the unit tests didn't catch.

Reply:
I think this might be related to one of my recent PRs, where I addressed the unusual use of the spread operator, which could also be causing the OOM issue. It would be great to have your review and opinion, @larsrickert. Please go through my PR [#22994](https://github.com/storybookjs/storybook/pull/22994). @et84121, your opinion is much welcome as well.

Reply:
Otherwise, with `vue-component-meta`, we have an option to ignore types using the schema: { ignore: ['MyIgnoredNestedProps'] }. For more details, see the [README.md](https://github.com/vuejs/language-tools/blob/master/packages/component-meta/README.md).

Reply:
@et84121 @chakAs3 I think I found a solution without needing to limit `vue-component-meta` features.
Please check out #28589

Reply:
Thank you both, @larsrickert and @chakAs3, for your thorough investigation and insights.

@larsrickert, I appreciate your deep dive into the issue. Your solution of addressing the problem within Storybook's processing, rather than limiting vue-component-meta's capabilities, is indeed a much better approach. I agree that we shouldn't restrict vue-component-meta's functionality, as it could lead to unintended consequences in other projects. I apologize for not investigating the root cause more thoroughly before proposing my solution.

@chakAs3, thank you for sharing your perspective and referencing your previous PR. While I'm not certain if the circular reference issue is directly related to the root cause here, your input is valuable in considering all potential factors.

Given these developments, I believe it would be best to close this PR and proceed with @larsrickert's solution in PR #28589. I'm glad that Storybook is addressing this issue, as it will greatly help me in advocating for Storybook's adoption as a Component Document tool in my current project.

Thank you all for your collaborative efforts in resolving this. If there's anything else I can do to help, please let me know.